internal/infrastructure/kubernetes/proxy/resource.go

```diff
@@ -119,12 +120,15 @@ func expectedProxyContainers(infra *ir.ProxyInfra,
 				return nil, fmt.Errorf("invalid protocol %q", p.Protocol)
 			}
 			port := corev1.ContainerPort{
-				// hashed container port name including up to the 6 characters of the port name and the maximum of 15 characters.
```

Comment:
@cnvergence can you recall why this hash was added ? I think it was to limit the `name` length, so we want want to change the args we pass, but we still may need to use `GetHashedName`

Reply:
the containerport name should not matter unless there is a kubernetes service which will refer to that name. https://github.com/kubernetes/api/blob/master/core/v1/types.go#L2127-L2130

However, in case of envoy the service (type: loadbalancer) are using `targetPort` as Number instead of name. So the name is not used at all anywhere by default. https://github.com/kubernetes/api/blob/master/core/v1/types.go#L5410-L5418

I do not see why we should still use `GetHashedName`. The only reason for that would be that name is going to be too long, but I cannot see that case really. If the name is [protocol]-[portnumber] the maximum lenght is going to be `SCTP-66666` which equals to 10. That is not too long.

Reply:
We added the hash because we used listener port names as the container port names which could be larger than 15 chars
and in merged gateways we used the format with namespace/gateway/listener

Reply:
this would be a good change for any deployment mode

Reply:
* can we instead fix the portName here https://github.com/envoyproxy/gateway/blob/7f5cbe84a2332a9f6b194ff2c276a320354a55f7/internal/gatewayapi/listener.go#L146 in the gatewayapi layer and reuse `p.Name` here ?
we'll need to make sure its not being used anywhere else

* and can we define the infraIRListenerPortName() https://github.com/envoyproxy/gateway/blob/7f5cbe84a2332a9f6b194ff2c276a320354a55f7/internal/gatewayapi/helpers.go#L375

sorry for nits, but trying to keep the code simpler to maintain